Moved configuring to service provider. Added post() method for raw payload.

dkulyk opened the pull request:
GA4MeasurementProtocol required configuration for using. This is a problem for using it manually without package configuring.

Also need to send additional user parameters with events like user_id.

gauravmak commented on src/GA4ServiceProvider.php:
```diff
@@ -21,14 +22,31 @@ public function configurePackage(Package $package): void
     {
         $this->app->bind('ga4', function () {
-            return new GA4MeasurementProtocol();
+            if (config('google-analytics-4-measurement-protocol.measurement_id') === null
+                || config('google-analytics-4-measurement-protocol.api_secret') === null
+            ) {
+                throw new Exception('Please set .env variables for Google Analytics 4 Measurement Protocol as per the readme file first.');
+            }
```
Does this mean that someone will face this exception for any request just after installing the package until they set the .env vars?

dkulyk replied:
This called only on resolving 'ga4'. Same as in the constructor.

gauravmak commented on src/GA4ServiceProvider.php:
```diff
@@ -21,14 +22,31 @@ public function configurePackage(Package $package): void
+            return new GA4MeasurementProtocol(
+                config('google-analytics-4-measurement-protocol.measurement_id'),
+                config('google-analytics-4-measurement-protocol.api_secret'),
+                function () {
+                    $clientId = session(config('google-analytics-4-measurement-protocol.client_id_session_key'));
+                    if (empty($clientId)) {
+                        throw new Exception('Please use the package provided blade directive or set client_id manually before posting an event.');
+                    }
+
+                    return $clientId;
+                }
```
What is the benefit of passing the closure from here instead of keeping the login inside the facade?

dkulyk replied:
Service class should not  know about configuration. But provider must configuring service.

gauravmak commented:
Thanks for the PR Dmytro.

> Also need to send additional user parameters with events like user_id.

Maybe we can have a separate PR for this to keep things simple?

gauravmak commented:
Thanks for the PR Dmytro.

> Also need to send additional user parameters with events like user_id.

Maybe we can have a separate PR for this to keep things simple?

dkulyk commented:
Reverted "Also need to send additional user parameters with events like user_id."

Simplified  using only clientId for allow Closure for dynamic resolver for clientId

gauravmak commented:
A bit occupied these days. Will revert after 17th May, thanks.

gauravmak commented:
@dkulyk I haven't added any tests in the package yet. Reason - I wasn't sure how to.

Do you have any ideas?
